framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/WorkbenchWithKieServerInDifferentNamespaceScenarioImpl.java

```diff
@@ -0,0 +1,186 @@
+        return workbenchProjectName;
+    }
+
+    @Override
+    public void deploy() {
+        // OpenShift restriction: Hostname must be shorter than 63 characters
+        workbenchProjectName = UUID.randomUUID().toString().substring(0, 4);
+        OpenShiftConstants.getNamespacePrefix().ifPresent(p -> workbenchProjectName = p + "-" + workbenchProjectName);
+
+        logger.info("Generated Workbench project name is " + workbenchProjectName);
+
+        kieServerProjectName = UUID.randomUUID().toString().substring(0, 4);
+        OpenShiftConstants.getNamespacePrefix().ifPresent(p -> kieServerProjectName = p + "-" + kieServerProjectName);
+
+        logger.info("Generated Kie server project name is " + kieServerProjectName);
+
+        logger.info("Creating project " + workbenchProjectName);
+        Project workbenchProject = openshiftController.createProject(workbenchProjectName);
+
+        logger.info("Creating project " + kieServerProjectName);
+        Project kieServerProject = openshiftController.createProject(kieServerProjectName);
+
+        logger.info("Creating image streams from " + OpenShiftConstants.getKieImageStreams());
+        workbenchProject.createResources(OpenShiftConstants.getKieImageStreams());
+        kieServerProject.createResources(OpenShiftConstants.getKieImageStreams());
+
+        logger.info("Processing template and creating resources from " + OpenShiftConstants.getKieAppTemplateWorkbench());
+        envVariables.put(OpenShiftTemplateConstants.IMAGE_STREAM_NAMESPACE, workbenchProjectName);
+        workbenchProject.processTemplateAndCreateResources(OpenShiftConstants.getKieAppTemplateWorkbench(), envVariables);
+
+        logger.info("Processing template and creating resources from " + OpenShiftConstants.getKieAppTemplateKieServerDatabase());
+        envVariables.put(OpenShiftTemplateConstants.IMAGE_STREAM_NAMESPACE, kieServerProjectName);
+        kieServerProject.processTemplateAndCreateResources(OpenShiftConstants.getKieAppTemplateKieServerDatabase(), envVariables);
+
+        workbenchDeployment = new WorkbenchDeploymentImpl();
+        workbenchDeployment.setOpenShiftController(openshiftController);
+        workbenchDeployment.setNamespace(workbenchProjectName);
+        workbenchDeployment.setUsername(DeploymentConstants.getWorkbenchUser());
+        workbenchDeployment.setPassword(DeploymentConstants.getWorkbenchPassword());
+        workbenchDeployment.setServiceName(OpenShiftConstants.getKieApplicationName());
+
+        String routeHostWorkbench = workbenchProject.getService(workbenchDeployment.getServiceName()).getRoute().getRouteHost();
+        String urlWorkbench = "http://" + routeHostWorkbench;
+        try {
+            workbenchDeployment.setUrl(new URL(urlWorkbench));
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Malformed URL for workbench", e);
+        }
+
+        kieServerDeployment = new KieServerDeploymentImpl();
```

Comment:
This method is kind of big. Could you please split it somehow (either two methods one for each project or three - one for each deployment?)